core/types/calls.cc

```diff
@@ -832,6 +832,8 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, const DispatchArgs &arg
     }
     auto posArgs = args.numPosArgs;
     bool hasKwargs = absl::c_any_of(data->arguments, [](const auto &arg) { return arg.flags.isKeyword; });
```

Comment:
Do we need to audit any of our other uses of `hasKwargs` to see if it should be using `hasNonDefaultKwParams` instead?

Reply:
I've double checked that and didn't find any obvious spots to update. 

The edge case I was trying to fix is limited to the positional arguments dispatching. The code responsible for this ends with the loop I was changing. The rest of the method is about dispatching keyword args, so the change shouldn't affect it.

Additionally to that, CI has passed and typecheck over p-s didn't show any errors